### PR TITLE
stats: Ensure that events are aggregated only once

### PR DESCRIPTION
**Pull request by psaiz**
:heart: Thank you for your contribution!

### Description

Ensures that a second aggregation does not reprocess any aggregation already performed and that has not changed,

It addresses https://github.com/zenodo/rdm-project/issues/523

* closes https://github.com/zenodo/rdm-project/issues/523

### CODE REVIEW

File: invenio_stats/aggregations.py [hunk 1]
```diff
@@ -225,10 +225,9 @@ def agg_iter(self, dt, previous_bookmark):
         )
         if previous_bookmark:
             # Let's make sure that both objects are of the same type. This is important for the test
-            previous = float(previous_bookmark.strftime("%s"))
+            previous = previous_bookmark.isoformat()
```
Comment (ntarocco) [inline invenio_stats/aggregations.py:228]: @psaiz this is not a string, but down there is evaluated as a number when comparing with `<`. Are you sure that this is correct?
Are the tests catching this?

Comment (psaiz) [reply invenio_stats/aggregations.py:228]: Thanks for looking into this. It is a string, isn't it? Note that the comparison is with strings.

I've added these prints to make sure:
```
                 ).replace(**dict.fromkeys(INTERVAL_ROUNDING[self.interval], 0))
+                print("Comparing")
+                print(aggregation["last_update"]["value_as_string"])
+                print(previous)
                 if aggregation["last_update"]["value_as_string"] and previous:
                     if aggregation["last_update"]["value_as_string"] < previous:
+                        print("SKIPPING EVENT")
                         continue
```

And the test seems to be doing it properly:
```
Comparing
2017-06-02T11:00:00.000Z
None
Comparing
2017-06-02T11:00:00.000Z
None
Comparing
2017-07-01T05:00:00.000Z
2017-06-02T11:59:00
Comparing
2017-07-01T05:00:00.000Z
2017-06-02T11:59:00
Comparing
2017-07-01T05:00:00.000Z
2017-07-01T05:59:00
SKIPPING EVENT
```

Comment (ntarocco) [reply invenio_stats/aggregations.py:228]: Sorry, I wanted to write `it is now a string`, but I wrote it is `not` a string, I am sorry :)
Do we really want to compare strings in this way? 
https://docs.python.org/3/tutorial/datastructures.html#comparing-sequences-and-other-types

I would feel much safer if we convert everything first with `datetime.fromisoformat()`, and then we use the Python date comparison. WDYT?

Comment (psaiz) [reply invenio_stats/aggregations.py]: Ok, I have no strong opinions about comparing strings or dates. The added value of comparing the dates is that there is no need to modify the incoming object.

Do you prefer this approach?

**Review by ntarocco — APPROVED** [review]

**Pull request merged**